src/ring_theory/free_comm_ring.lean

```diff
@@ -0,0 +1,270 @@
+/-
+Copyright (c) 2019 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import group_theory.free_abelian_group ring_theory.subring data.set.finite data.real.irrational
+
+universes u v
+
+def free_comm_ring (α : Type u) : Type u :=
+free_abelian_group $ multiset α
+
```

Comment:
I don't think this should end up in mathlib. If you feel that strongly about decidability assumptions refactor polynomials.

Reply:
If I use polynomial instead of this, where should I put the things about `supported`?

Reply:
I can't really work with mv_polynomial. It is really slow.

Reply:
It is clarified in https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/free_comm_ring that `free_comm_ring` is ok. @ChrisHughes24

Reply:
I'd prefer that this part is in a separate PR, but if Chris knows what's going on here I'm fine with him making the call.

Reply:
By "separate PR" do you mean #734?